classes/mock/generator.php

```diff
@@ -317,6 +325,11 @@ protected function generateClassMethodCode(\reflectionClass $class)
 			$mockedMethods .= PHP_EOL;
 			$mockedMethods .= "\t" . '{' . PHP_EOL;
 
+			if ($this->eachInstanceIsUnique === true)
+			{
+				$mockedMethods .= "\t\t" . '$this->id = uniqid();' . PHP_EOL;
```

Comment:
A solution would be to generate a randomly named attribute:
```php
$this->{uniqid()} = true
```
Be careful that `uniqid()` must compute a valid PHP identifier. So maybe:
```php
$this->___atoum_{uniqid()} = true
```
Triple-underscore should be enought to avoid potential collision.